mtags/src/main/scala-3/scala/meta/internal/pc/InferredTypeProvider.scala

```diff
@@ -236,7 +241,10 @@ final class InferredTypeProvider(
        */
       case Some(df @ DefDef(name, _, tpt, rhs)) =>
         def typeNameEdit =
-          val end = tpt.endPos.toLsp
+          val end =
+            if tpt.endPos.`end` > df.namePos.`end` then tpt.endPos.toLsp
```

Comment:
```suggestion
            if tpt.endPos.end > df.namePos.end then tpt.endPos.toLsp
```
Looks like an issue with escaping soft keywords, we shouldn't escaped them if used in scope completions :thinking: 

Could you also explain the issue in the comment? And that it can be removed once we remove support for 3.1.3 at some point in the future?

Reply:
In Scala 3.1.3
```scala
List((1,2)).map((<<a>>,b) => a)
// would insert type here
List((1,2)).map((:Inta,b) => a)
```

Reply:
I just meant in the comment in the code sweat_smile